server/src/controllers/user.controller.ts

```diff
@@ -0,0 +1,302 @@
+        );
+        res.json([registeredUser[0], access[0]]);
+    } catch (err) {
+        next(err);
+    }
+};
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+const updateItemAndVisibility = async (req: any, res: Response, next: NextFunction): Promise<void> => {
+    try {
+        const user = await RegisteredUser.findOneAndUpdate(
+            { email: req.user.email },
+            req.body,
+            { new: true, useFindAndModify: false }
+        );
+        const key = Object.keys(req.body)[0] as RegisteredUserKey;
+        user ? res.json({ [key]: user[key] }) : next();
+    } catch (err) {
+        next(err);
+    }
+};
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+const uploadProfileImg = async (req: any, res: Response, next: NextFunction): Promise<void> => {
+    const form = new IncomingForm({ multiples: false });
+    let fileName = '';
+
+    form.parse(req, (err, fields, files) => {
+        if (err) {
+            log.error(err, fields, files);
+            next();
+        } else {
+            log.info('Image upload parse successful.');
+        }
+    });
+    form.on('field', (name, field) => {
+        log.warn('Fields are invalid at this URL.', name, field);
+    });
+    form.on('fileBegin', (name, file) => {
+        fileName = renameFile(file.name);
+        file.path = SERVER_ROOT + fileName;
+    });
+    form.on('file', async (name, file) => {
+        log.info(`File: ${file.name}, ${file.size} byte, ${file.type}`);
+        try {
+            const user = await RegisteredUser.findOneAndUpdate(
+                { email: req.user.email },
+                { profileImg: fileName },
+                { useFindAndModify: false }
+            );
+            res.json({ profileImg: fileName });
+            log.info(`User new profile image is: ${fileName}`);
+            if (user) {
+                const removeFile = SERVER_ROOT + user.profileImg; // former profile img
+                if (user.profileImg) {
+                    fs.unlink(removeFile, (err: NodeJS.ErrnoException | null) => {
+                        if (err) {
+                            log.warn(`Failed to delete profile image: ${removeFile}`);
+                            return;
+                        }
+                        log.info(`Removed profile image: ${removeFile}`);
+                    });
+                }
+            }
+        } catch (err) {
+            next(err);
+        }
+    });
+    form.on('aborted', () => {
+        log.info('Request aborted by the user');
+        res.status(500).send({ error: 'Request Aborted.' });
+    });
+    form.on('error', (err: Error) => {
+        next(err);
+    });
+    form.on('end', () => {
+        log.info('Upload successful.');
+    });
+};
+
+type MemberLevel = Pick<IMember, '_id' | 'level' | 'email'>;
+
+const findMemberLevel = (membersLevel: Array<MemberLevel>, registeredUser: IRegisteredUser): string => {
```

Comment:
It is enough passing registeredUser.email to this function. No need for the whole object